src/coreclr/nativeaot/Runtime/FinalizerHelpers.cpp

```diff
@@ -184,6 +186,11 @@ EXTERN_C UInt32_BOOL QCALLTYPE RhpWaitForFinalizerRequest()
 // Indicate that the current round of finalizations is complete.
 EXTERN_C void QCALLTYPE RhpSignalFinalizationComplete(uint32_t fcount)
 {
+    if (YieldProcessorNormalization::IsMeasurementScheduled())
```

Comment:
We can do this after `g_FinalizerDoneEvent`. ``g_FinalizerDoneEvent` can unblock the rest of the system. It is better to do it earlier rather than later.